packages/opentelemetry-basic-tracer/src/Span.ts

```diff
@@ -126,7 +132,10 @@ export class Span implements types.Span {
 
   private _isSpanEnded(): boolean {
     if (this._ended) {
-      // @todo: log a warning
+      this._logger.warn(
+        'Can not execute the operation on ended Span: %s.',
+        JSON.stringify(this._spanContext)
```

Comment:
This will be called even if the log level is not matched. Logging should not impact the performance of the service. We should have a way to avoid expensive calls such as creating errors and serializing when the log level is not matched.

Reply:
Agreed. For now, I have just removed `JSON.stringify()` in the log.